I tested this by changing the kos-ports/libmp3/Makefile to
# What files we need to download, and where from.
#GIT_REPOSITORIES =  git://git.code.sf.net/p/cadcdev/libmp3 \
#                    https://github.com/KallistiOS/libmp3.git
GIT_REPOSITORIES =  https://github.com/KallistiOS/libmp3.git
GIT_BRANCH =        stdint

but when I run make install , I get an error in sndmp3.c about implicit declaration of function memalign.
In file included from sndmp3.c:32:
../xingmp3/mhead.h:292:48: note: expected ‘unsigned char *’ but argument is of type ‘char *’
  292 |    IN_OUT audio_decode(MPEG *m, unsigned char *bs, short *pcm);
      |                                 ~~~~~~~~~~~~~~~^~
sndmp3.c: In function ‘xing_init’:
sndmp3.c:187:30: error: implicit declaration of function ‘memalign’ [-Wimplicit-function-declaration]
  187 |                 pcm_buffer = memalign(32, PCM_SIZE);
      |                              ^~~~~~~~
sndmp3.c:187:28: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  187 |                 pcm_buffer = memalign(32, PCM_SIZE);
      |                            ^

If I add a #include <malloc.h>
that will resolve the issue.

--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,6 @@
 
 TARGET = libmp3.a
 OBJS =
-LIB_OBJS = build/*.o
 SUBDIRS = xingmp3 libmp3
 LOCAL_CLEAN = build/*.o build/*.a
 # SUBDIRS = mpglib libmp3
@@ -15,6 +14,6 @@ include $(KOS_BASE)/addons/Makefile.prefab
 
 # creates the kos link to the headers
 create_kos_link:
-	rm -f ../include/mp3
-	ln -s ../libmp3/include ../include/mp3
+	rm -f $(KOS_PORTS)/include/mp3
+	ln -s $(CURDIR)/include $(KOS_PORTS)/include/mp3
 

--- a/include/sfxmgr.h
+++ b/include/sfxmgr.h
@@ -1,7 +1,7 @@
 /* KallistiOS ##version##
 
    sfxmgr.h
-   (c)2000 Dan Potter
+   (c)2000 Megan Potter
 */
 
 #ifndef __SFXMGR_H
@@ -12,7 +12,7 @@ __BEGIN_DECLS
 
 int sfx_load(const char *fn);
 void sfx_play(int idx, int vol, int pan);
-void sfx_unload_all();
+void sfx_unload_all(void);
 
 __END_DECLS
 

--- a/include/sndmp3.h
+++ b/include/sndmp3.h
@@ -1,7 +1,7 @@
 /* KallistiOS ##version##
 
    sndmp3.h
-   (c)2000 Dan Potter
+   (c)2000 Megan Potter
 */
 
 #ifndef __SNDMP3_H

--- a/include/sndserver.h
+++ b/include/sndserver.h
@@ -1,7 +1,7 @@
 /* KallistiOS ##version##
 
    sndserver.h
-   (c)2000 Dan Potter
+   (c)2000 Megan Potter
 */
 
 #ifndef __SNDSERVER_H
@@ -11,16 +11,16 @@
 __BEGIN_DECLS
 
 /* Starts the MP3 server thread */
-int mp3_init();
+int mp3_init(void);
 
 /* Starts a song playing */
 int mp3_start(const char *fn, int loop);
 
 /* Stops the playing song */
-int mp3_stop();
+int mp3_stop(void);
 
 /* Shuts down the MP3 server thread */
-int mp3_shutdown();
+int mp3_shutdown(void);
 
 /* Controls volume level of stream */
 void mp3_volume(int vol);

--- a/libmp3/Makefile
+++ b/libmp3/Makefile
@@ -1,7 +1,7 @@
 # KallistiOS ##version##
 #
 # libmp3/Makefile
-# (c)2001 Dan Potter
+# (c)2001 Megan Potter
 
 OBJS = main.o sndmp3.o
 

--- a/libmp3/main.c
+++ b/libmp3/main.c
@@ -1,11 +1,13 @@
 /* KallistiOS ##version##
 
    libmp3 main.c
-   (c)2000-2001 Dan Potter
+   (c)2000-2001 Megan Potter
 
 */
 
-#include <kos.h>
+#include <stdio.h>
+#include <kos/thread.h>
+#include <dc/sound/stream.h>
 #include "sndmp3.h"
 
 void *sndserver_thread(void *blagh) {
@@ -18,7 +20,7 @@ void *sndserver_thread(void *blagh) {
 	return NULL;
 }
 
-int mp3_init() {
+int mp3_init(void) {
 	if (snd_stream_init() < 0)
 		return -1;
 	if (thd_create(1, sndserver_thread, NULL) != NULL) {

--- a/libmp3/sndmp3.c
+++ b/libmp3/sndmp3.c
@@ -1,7 +1,7 @@
 /* KallistiOS ##version##
 
    sndmp3.c
-   Copyright (C)2000,2004 Dan Potter
+   Copyright (C)2000,2004 Megan Potter
 
    An MP3 player using sndstream and XingMP3
 */
@@ -13,10 +13,21 @@
 
  */
 
-#include <kos.h>
+
 #include <assert.h>
+#include <malloc.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
 #include "sndserver.h"
 #include "sndmp3.h"
+
+#include <dc/spu.h>
+#include <dc/sound/stream.h>
+#include <kos/sem.h>
+#include <kos/thread.h>
 
 /************************************************************************/
 #include <mhead.h>		/* From xingmp3 */
@@ -49,7 +60,7 @@ static int pcm_count, pcm_discard;
 static snd_stream_hnd_t stream_hnd = -1;
 
 /* MPEG file */
-static uint32 mp3_fd;
+static uint32_t mp3_fd;
 static int		frame_bytes;
 static MPEG		mpeg;
 static MPEG_HEAD	head;
@@ -62,7 +73,7 @@ static char mp3_last_fn[256];
 /* Checks to make sure we have some data available in the bitstream
    buffer; if there's less than a certain "water level", shift the
    data back and bring in some more. */
-static int bs_fill() {
+static int bs_fill(void) {
 	int n;
 
 	/* Make sure we don't underflow */
@@ -150,7 +161,7 @@ errorout:
 
 /* Open an MPEG stream and prepare for decode */
 static int xing_init(const char *fn) {
-	uint32	fd;
+	uint32_t	fd;
 
 	/* Open the file */
 	mp3_fd = fd = fs_open(fn, O_RDONLY);
@@ -201,7 +212,7 @@ static int xing_init(const char *fn) {
 		printf("Final index is %d\r\n", (bs_ptr - bs_buffer));
 	}
 
-	if (((uint8)bs_ptr[0] != 0xff) && (!((uint8)bs_ptr[1] & 0xe0))) {
+	if (((uint8_t)bs_ptr[0] != 0xff) && (!((uint8_t)bs_ptr[1] & 0xe0))) {
 		printf("Definitely not an MPEG file\r\n");
 		goto errorout;
 	}
@@ -280,8 +291,6 @@ static void xing_shutdown() {
 
 
 /************************************************************************/
-
-#include <dc/sound/stream.h>
 
 /* Status flag */
 #define STATUS_INIT	0

--- a/libmp3/sndmp3_mpglib.c
+++ b/libmp3/sndmp3_mpglib.c
@@ -1,7 +1,7 @@
 /* Tryptonite
 
    sndmp3.c
-   (c)2000 Dan Potter
+   (c)2000 Megan Potter
 
    An MP3 player using sndstream and MPGLIB
 
@@ -37,7 +37,7 @@ static char *pcm_buffer = NULL, *pcm_ptr;
 static int pcm_count;
 
 /* MPEG file */
-static uint32		mp3_fd;
+static uint32_t		mp3_fd;
 static struct mpstr	mp;
 
 /* Name of last played MP3 file */
@@ -46,7 +46,7 @@ static char mp3_last_fn[256];
 /* Checks to make sure we have some data available in the bitstream
    buffer; if there's less than a certain "water level", shift the
    data back and bring in some more. */
-static int bs_fill() {
+static int bs_fill(void) {
 	int n;
 
 	/* Read in some more data */
@@ -132,7 +132,7 @@ errorout:
 /* Open an MPEG stream and prepare for decode */
 static int mpglib_init(const char *fn) {
 	int	size;
-	uint32	fd;
+	uint32_t	fd;
 
 	/* Open the file */
 	mp3_fd = fd = fs_open(fn, O_RDONLY);
@@ -183,7 +183,7 @@ static int mpglib_init(const char *fn) {
 		printf("Final index is %d\r\n", (bs_ptr - bs_buffer));
 	}
 
-	if (((uint8)bs_ptr[0] != 0xff) && (!((uint8)bs_ptr[1] & 0xe0))) {
+	if (((uint8_t)bs_ptr[0] != 0xff) && (!((uint8_t)bs_ptr[1] & 0xe0))) {
 		printf("Definitely not an MPEG file\r\n");
 		goto errorout;
 	}
@@ -250,7 +250,7 @@ static void mpglib_shutdown() {
 static volatile int sndmp3_status;
 
 /* Wait until the MP3 thread is started and ready */
-void sndmp3_wait_start() {
+void sndmp3_wait_start(void) {
 	while (sndmp3_status != STATUS_READY)
 		;
 }
@@ -263,7 +263,7 @@ static volatile int sndmp3_loop;
 
 /* Call this function as a thread to handle playback. Playback will 
    stop and this thread will return when you call sndmp3_shutdown(). */
-static void sndmp3_thread() {
+static void sndmp3_thread(void) {
 	int sj;
 	
 	/* Main command loop */
@@ -357,7 +357,7 @@ int sndmp3_start(const char *fn, int loop) {
 }
 
 /* Stop playback (implies song unload) */
-void sndmp3_stop() {
+void sndmp3_stop(void) {
 	if (sndmp3_status == STATUS_READY)
 		return;
 	sndmp3_status = STATUS_STOPPING;
@@ -368,7 +368,7 @@ void sndmp3_stop() {
 }
 
 /* Shutdown the player */
-void sndmp3_shutdown() {
+void sndmp3_shutdown(void) {
 	sndmp3_status = STATUS_QUIT;
 	sem_signal(sndmp3_halt_sem);
 	while (sndmp3_status != STATUS_ZOMBIE)
@@ -386,7 +386,7 @@ void sndmp3_volume(int vol) {
 #include "sfxmgr.h"
 #include "sndsrv_abi.h"
 static abi_sndsrv_t ssabi;
-static void sndmp3_svc_init() {
+static void sndmp3_svc_init(void) {
 	memset(&ssabi, 0, sizeof(ssabi));
 	ssabi.hdr.version = ABI_MAKE_VER(1,0,0);
 
@@ -404,7 +404,7 @@ static void sndmp3_svc_init() {
 } */
 
 /* The main loop for the sound server */
-void sndmp3_mainloop() {
+void sndmp3_mainloop(void) {
 	/* Allocate a semaphore for temporarily halting sndmp3 */
 	sndmp3_halt_sem = sem_create(0);
 

